Desctiption: use packaged version of libann instead of shipped with tarball
Author: Christophe Trophime <christophe.trophime@lncmi.cnrs.fr>
Reviewed-by: Anton Gladky <gladky.anton@gmail.com>
Last-Update: 2012-01-16

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -451,9 +451,19 @@
 endif(HAVE_FLTK OR HAVE_QT OR ENABLE_GRAPHICS)
 
 if(ENABLE_ANN)
-  add_subdirectory(contrib/ANN)
-  include_directories(contrib/ANN/include)
-  set_config_option(HAVE_ANN "Ann")
+    find_library(ANN_LIB ann PATH_SUFFIXES lib)
+    if(ANN_LIB)
+      find_path(ANN_INC "ANN.h" PATH_SUFFIXES src include ANN)
+      if(ANN_INC)
+        list(APPEND EXTERNAL_LIBRARIES ${ANN_LIB})
+        list(APPEND EXTERNAL_INCLUDES ${ANN_INC})
+      endif(ANN_INC)
+    else(ANN_LIB)
+       message(STATUS "ANN " ${ANN_LIB} " not Found - using contrib/ann instead")
+       add_subdirectory(contrib/ANN)
+       include_directories(contrib/ANN/include)
+    endif(ANN_LIB)
+    set_config_option(HAVE_ANN "Ann")
 endif(ENABLE_ANN)
 
 if(ENABLE_BLOSSOM)
